src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -323,6 +333,7 @@ impl ThinPool {
     /// metadata save has been made.
     pub fn check(&mut self, dm: &DM, backstore: &mut Backstore) -> EngineResult<bool> {
         #![allow(match_same_arms)]
+        assert_eq!(backstore.device(), self.backstore_device);
```

Comment:
Forward-looking. Is correct within current context, but would seem unduly paranoid w/out expected changes.